### ISSUE TITLE
the generation code of train dataset.

### ISSUE THREAD

**Issue by mynameischaos** [HaozheZhao/MIC#9]
Hi haozhe,

MIC is nice work!

Can you provide the generation code of the training dataset? We want to follow your method to generate more datasets.

Best wishes.

**Comment by HaozheZhao** [comment]
Of course, we will provide our codebase for data generation. In fact, we consider this to be a very important part of our work. The code base will be released within **two weeks**, along with the paper presenting the tool for easily creating a MIC dataset from the open source dataset.

**Comment by mynameischaos** [comment]
> Of course, we will provide our codebase for data generation. In fact, we consider this to be a very important part of our work. The code base will be released within **two weeks**, along with the paper presenting the tool for easily creating a MIC dataset from the open source dataset.

Hi haozhe，

May I ask if it is convenient to provide this code？Thank you very much.

**Comment by HaozheZhao** [comment]
> > Of course, we will provide our codebase for data generation. In fact, we consider this to be a very important part of our work. The code base will be released within **two weeks**, along with the paper presenting the tool for easily creating a MIC dataset from the open source dataset.
> 
> Hi haozhe，
> 
> May I ask if it is convenient to provide this code？Thank you very much.

Hi, we are managing to generate more data from different datasources with our code base. We have manually designed input and output templates for various datasets. We've created approximately 7 million new pieces of data and selected 1.5 million for tuning. We aim to provide you with the high quality data, and thus, we plan to make the code available once we've confirmed the performance of the newly generated data with ICL-tuning. Furthermore, our code base is somewhat messy at the moment, but rest assured, we are working on cleaning it up. 

Thanks for you attention about our work. I will manage to release the code ASAP.
